odoo/addons/base/models/ir_rule.py

```diff
@@ -68,6 +68,18 @@ def _check_model_name(self):
         if any(rule.model_id.model == self._name for rule in self):
             raise ValidationError(_('Rules can not be applied on the Record Rules model.'))
 
+    @api.constrains('active', 'domain_force', 'model_id')
+    def _check_domain(self):
+        eval_context = self._eval_context()
+        for rule in self:
+            if rule.active and rule.domain_force and rule.model_id:
+                try:
+                    # dummy search to make sure the domain is valid
+                    domain = safe_eval(rule.domain_force, eval_context)
+                    self.env[rule.model_id.model].sudo()._search(domain, limit=1)
```

Comment:
I made a mistake by proposing `_search`, there is a better approach (bypass useless override and make sure to test a falsy domain, see https://github.com/odoo/odoo/blob/2a71dbf937d0c2cb7843f10041616e24e4d157ef/odoo/models.py#L5439): 
`expression(domain, self.env[rule.model_id.model].sudo())`  (with `from odoo.osv.expression import expression`)